index.bs

```diff
@@ -47,66 +51,88 @@ dictionary CSSStyleSheetInit {
-		2. If the {{CSSStyleSheetInit/media}} attribute of <var>options</var> is a string,
+		2.	Set <var>sheet’s</var> origin-clean flag.
+			Set <var>sheet's</var> [=constructed flag=].
+			Set <var>sheet's</var> [=constructor document=] to the {{Document}} for the window where this constructor is called on.
+		3. If the {{CSSStyleSheetInit/media}} attribute of <var>options</var> is a string,
 			<a>create a MediaList object</a> from the string
 			and assign it as <var>sheet’s</var> media.
 			Otherwise, assign a copy of the value of the attribute as <var>sheet’s</var> media.
-		3. If the {{CSSStyleSheetInit/alternate}} attribute of <var>options</var> is true,
+		4. If the {{CSSStyleSheetInit/alternate}} attribute of <var>options</var> is true,
 			set <var>sheet’s</var> alternate flag.
-		4. If the {{CSSStyleSheetInit/disabled}} attribute of <var>options</var> is true,
+		5. If the {{CSSStyleSheetInit/disabled}} attribute of <var>options</var> is true,
 			set <var>sheet’s</var> disabled flag.
-		5. <a spec=css-syntax-3>Parse a stylesheet</a> from {{text}}.
+		6. Return <var>sheet</var>.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="constructed flag">constructed flag</dfn></dt>
+	<dd>
+		Specified when created. Either set or unset. Unset by default.
+		Signifies whether this stylesheet is made via constructor or not, so must be set only for stylesheets that are constructed using the {{CSSStyleSheet()}} or {{CSSStyleSheet(options)}} function.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="disallow modification flag">disallow modification flag</dfn></dt>
+	<dd>
+		Either set or unset. Unset by default. If set, modification to the stylesheet's rules are not allowed.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="constructor document">constructor document</dfn></dt>
+	<dd>
+		Specified when created. The {{Document}} where the CSSStyleSheet is originally constructed on. Null by default.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="set of adopter documents">set of adopter documents</dfn></dt>
+	<dd>
+		A set of {{Document}}s on which the stylesheet is adopted (e.g. there is at least one {{DocumentOrShadowRoot}} within that {{Document}}'s tree on which the sheet is part of the {{DocumentOrShadowRoot}}'s {{DocumentOrShadowRoot/adoptedStyleSheets}}).
+	</dd>
+</dl>
+
+Modifying Constructed Stylesheets {#modifying-constructed-stylesheets}
+=============================
+
+After construction, constructed stylesheets can be modified using rule modification methods like <a spec="cssom-1" for="CSSStyleSheet">insertRule(rule)</a> or <a spec="cssom-1" for="CSSStyleSheet">deleteRule(rule)</a>, or {{replace(text)}} and {{replaceSync(text)}} if the sheet's [=disallow modification flag=] is not set. If those methods are called when the sheet's [=disallow modification flag=] is set, or <a spec="cssom-1" for="CSSStyleSheet">insertRule(rule)</a> is used to add an <a spec=css-cascade-4>@import</a> rule, a "{{NotAllowedError}}" {{DOMException}} must be thrown. 
```

Comment:
This is OK, but it's a classic [monkey patch](https://annevankesteren.nl/2014/02/monkey-patch). As such we should make it clearer exactly what the patch is, that you would eventually change the CSSOM spec for. I'd suggest something like the following, although more precise and using the correct spec terms:

insertRule(rule)

1. If this sheet's disallow modification flag is set, throw....
1. If rule is an import rule and this sheet's constructed flag is set, throw...
1. (The rest of the algorithm remains as in CSSOM)

and the same for all other methods that are affected.

Reply:
I see, added those. Should I keep these sentences as summary?

Reply:
Keeping the sentences is good, but you should change them to say "will throw an exception" instead of "must throw an exception". "Must" is a requirement that implementers should implement, which is confusing because then you have the requirement in two places: both that sentence, and the algorithms themselves. Whereas "will" implies that it will happen because some other normative text, i.e. the below algorithms.

You could even say "will throw an exception, as detailed in the below algorithms".

A nit: you used "this sheet" in step 3 but _sheet_ in steps 1 and 2.

Reply:
Done!